.github/workflows/ci_e2e.yaml

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+name: Tests
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  schedule:
+    - cron: '53 0 * * SAT' # Every Saturday at 00:53 UTC
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install tox
+        # TODO: Consider replacing with custom image on self-hosted runner OR pinning version
+        run: python3 -m pip install tox
+      - name: Run linters
+        run: tox run -e lint
+
+  build:
+    name: Build charms
+    uses: canonical/data-platform-workflows/.github/workflows/build_charms_with_cache.yaml@v2
+
+  e2e-tests:
+    strategy:
+      fail-fast: false
+      max-parallel: 1
+      matrix:
+        tox-environments:
+          - integration-e2e-basic-flow
+          - integration-e2e-password-rotation
+        options:
+          - 
+          - --tls
+          - --integrator
+          - --tls --integrator
+    name: ${{ matrix.tox-environments }}-${{matrix.options}}
+    needs:
+      - lint
+      - build
+    runs-on: Ubuntu_4C_16G
```

Comment:
**question:** What, if any, issues did you see with the tests on this machine? Did we need to fiddle with timeouts, or did it all work straight away?

Reply:
it should work straight away